opal/threads/wait_sync.c

```diff
@@ -63,7 +65,7 @@ int ompi_sync_wait_mt(ompi_wait_sync_t *sync)
      *  - our sync has been triggered.
      */
  check_status:
-    if( sync != wait_sync_list ) {
+    if( sync != wait_sync_list && num_thread_in_progress >= opal_max_thread_in_progress) {
```

Comment:
Somehow this condition added a lot of latency. (from 3us to 50 us ?!?)
Investigating....